### PR TITLE
refactor(ivy): remove dependency of Injector on Ivy

### PR DESCRIPTION
**Pull request by mhevery**
This change is a prerequasity for a later change which will turn the
'di' into its own bazel package. In order to do that we have to:
- Remove the dependency of `class Injector` on Ivy so that it can be
  compiled in isolation. We do that by using `-1` as special value for
  `__NG_ELEMENT_ID__` which tells the Ivy `NodeInjector` than
  `Injector` is being requested.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by IgorMinar — APPROVED** [review]
lgtm

File: packages/core/src/render3/di.ts [hunk 1]
```diff
@@ -315,6 +315,10 @@ export function getOrCreateInjectable<T>(
         setTNodeAndViewData(savePreviousOrParentTNode, saveLView);
       }
     } else if (typeof bloomHash == 'number') {
+      if (bloomHash === -1) {
+        // `-1` is a special value used to identify `Injector` types.
```
Comment (ocombe) [inline packages/core/src/render3/di.ts]: Could you explain this in the API docs of the function?
Also could you add API docs for this function? :D

File: packages/core/src/render3/di.ts [hunk 2]
```diff
@@ -536,7 +540,8 @@ export function bloomHashBitOrFactory(token: Type<any>| InjectionToken<any>| str
     return token.charCodeAt(0) || 0;
   }
   const tokenId: number|undefined = (token as any)[NG_ELEMENT_ID];
-  return typeof tokenId === 'number' ? tokenId & BLOOM_MASK : tokenId;
+  // Negative token IDs are used for special objects such as `Injector`
```
Comment (ocombe) [inline packages/core/src/render3/di.ts]: Could you explain this in the API docs of the function?

File: packages/core/src/di/interface/injector.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  // TODO(alxhub): make this 'const' when ngc no longer writes exports of it into ngfactory files.
+
+  Default = 0b0000,
+
+  /**
+   * Specifies that an injector should retrieve a dependency from any injector until reaching the
+   * host element of the current component. (Only used with Element Injector)
+   */
+  Host = 0b0001,
+  /** Don't descend into ancestors of the node requesting injection. */
```
Comment (kara) [inline packages/core/src/di/interface/injector.ts]:
```suggestion
  /** Don't ascend into ancestors of the node requesting injection. */
```

**Review by kara — APPROVED** [review]
LGTM, just a few nits

File: packages/core/src/render3/di.ts (hunk 1 above)
Comment (mhevery) [reply packages/core/src/render3/di.ts]: fixed

File: packages/core/src/render3/di.ts (hunk 2 above)
Comment (mhevery) [reply packages/core/src/render3/di.ts]: fixed

**Pull request closed without merging**